recipes/libdrm/all/conanfile.py

```diff
@@ -54,100 +59,91 @@ class LibdrmConan(ConanFile):
         "etnaviv": False,
         "valgrind": False,
         "freedreno-kgsl": False,
-        "udev": False
+        "udev": False,
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    def build_requirements(self):
-        self.build_requires("meson/0.64.1")
-
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
         if Version(self.version) >= "2.4.111":
             del self.options.libkms
 
     def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.intel:
-            self.requires("libpciaccess/0.16")
-        if self.settings.os == "Linux":
-            self.requires("linux-headers-generic/5.14.9")
+            self.requires("libpciaccess/0.17")
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("linux-headers-generic/5.15.128")
 
     def validate(self):
         if self.settings.os not in ["Linux", "FreeBSD"]:
             raise ConanInvalidConfiguration("libdrm supports only Linux or FreeBSD")
 
+    def build_requirements(self):
+        self.tool_requires("meson/1.2.1")
```

Comment:
You need to add `pkgconf` as a `tool_requires`.

```suggestion
    def build_requirements(self):
        self.tool_requires("meson/1.2.1")
        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
            self.tool_requires("pkgconf/2.0.3")
```

Reply:
Thanks!